TeamCode/src/main/java/org/firstinspires/ftc/teamcode/hardware/vision/ComputerVisionProcessor.java

```diff
@@ -50,57 +49,68 @@ public Object processFrame(Mat inputFrameRGB, long captureTimeNanos) {
         }
-        leftMat.update(inputFrameRGB);
         middleMat.update(inputFrameRGB);
+        rightMat.update(inputFrameRGB);
         framesProcessed++;
-        setPropLocation();
-        return inputFrameRGB; // Don't think this does anything
+        cameraTelemetry();
+        detectPropLocation();
+
+
+        return inputFrameRGB; // This return value is userContext in call to onDrawFrame()
     }
 
-    public void setPropLocation() {
-        if (propPosition != null) {
-            propFound = true;
+    public void detectPropLocation() {
+        if (framesProcessed < 10) {
             return;
         }
-        if (framesProcessed > 50) {
-            propPosition = PropPosition.RIGHT;
+        // If we have found the prop position before, do nothing
+        // If we have already looked at X frames, assume prop
+        // is on the right
+        if (framesProcessed > 60 && propPosition == null) {
+            propPosition = PropPosition.LEFT;
+            telemetry.addLine("No prop detected in 50 frames: GUESSING LEFT");
+            return;
         }
+
         if (allianceColor == AllianceColor.BLUE) {
-            if (leftMat.blueAmount > middleMat.blueAmount) {
-                propPosition = PropPosition.LEFT;
-            } else if (middleMat.blueAmount > leftMat.blueAmount) {
+            if (middleMat.blueAmount > PIXEL_THRESHOLD_CONSTANT && middleMat.blueAmount > rightMat.blueAmount) {
                 propPosition = PropPosition.MIDDLE;
+            } else if (rightMat.blueAmount > PIXEL_THRESHOLD_CONSTANT && rightMat.blueAmount > middleMat.blueAmount) {
+                propPosition = PropPosition.RIGHT;
             }
-        } else if (allianceColor == AllianceColor.RED){
-            if (leftMat.redAmount > middleMat.redAmount) {
-                propPosition = PropPosition.LEFT;
-            } else if (middleMat.redAmount > leftMat.redAmount) {
+        } else if (allianceColor == AllianceColor.RED) {
+            if (middleMat.redAmount > PIXEL_THRESHOLD_CONSTANT && middleMat.redAmount > rightMat.redAmount) {
                 propPosition = PropPosition.MIDDLE;
+            } else if (rightMat.redAmount > PIXEL_THRESHOLD_CONSTANT && rightMat.redAmount > middleMat.redAmount) {
+                propPosition = PropPosition.RIGHT;
             }
         } else {
-            throw new RuntimeException("There should be a team color defined at this point");
+            throw new RuntimeException("Alliance color must be defined");
         }
     }
+        @Override
+        public void onDrawFrame (Canvas canvas,int onscreenWidth, int onscreenHeight,
+        float scaleBmpPxToCanvasPx, float scaleCanvasDensity, Object userContext){
+            double scaleFactor = Math.min(
+                    canvas.getWidth() / targetSize.width,
+                    canvas.getHeight() / targetSize.height);
+            android.graphics.Rect rightRect = rightMat.createAndroidRect(scaleFactor);
```

Comment:
Also the default Paint is filled black. But you can do somethign else like setStyle(STROKE) so that it's just the border, and setColor(YELLOW) so that it's not black.